File: DatabaseHelper.java
package com.my.toko;

import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import android.util.Log;
import org.json.JSONArray;
import org.json.JSONObject;

public class DatabaseHelper extends SQLiteOpenHelper {

  private static final String DATABASE_NAME = "tokokasir.db";
  private static final int DATABASE_VERSION = 1;

  public DatabaseHelper(Context context) {
    super(context, DATABASE_NAME, null, DATABASE_VERSION);
  }

  @Override
  public void onCreate(SQLiteDatabase db) {
    // Buat tabel
    String sql =
      "create table produk (np INTEGER PRIMARY KEY AUTOINCREMENT, nama text, harga_beli numeric, harga_jual numeric);";
    Log.d("DATA", sql);
    db.execSQL(sql);

    // Insert data
    sql =
      "INSERT INTO produk (nama, harga_beli, harga_jual) VALUES ('Permen Mint', 200, 500);";
    db.execSQL(sql);
  }

  @Override
  public void onUpgrade(SQLiteDatabase db, int i, int i1) {
    db.execSQL("DROP TABLE IF EXISTS 'produk'");
    onCreate(db);
  }

  public String siud(String query) {
    SQLiteDatabase db = this.getWritableDatabase();
    try {
      db.execSQL(query);
      return "OK";
    } catch (Exception e) {
      return "";
    }
  }

  public String getData(String table) {
    String selectQuery = "SELECT * FROM " + table;
    SQLiteDatabase db = this.getReadableDatabase();
    Cursor c = db.rawQuery(selectQuery, null);

    JSONArray jsonArray = new JSONArray();
    if (c.moveToFirst()) {
      Integer ncol = c.getColumnCount();
      do {
        JSONObject row = new JSONObject();
        for (int i = 0; i < ncol; i++) {
          if (c.getColumnName(i) != null) {
            try {
              if (c.getString(i) != null) {
                row.put(c.getColumnName(i), c.getString(i));
              } else {
                row.put(c.getColumnName(i), "");
              }
            } catch (Exception e) {}
          } // end if
        }
        jsonArray.put(row);
      } while (c.moveToNext());
    }
    return jsonArray.toString();
  }
}

File: DatabaseHelper.kt
package com.databasekotlin

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import org.json.JSONArray
import org.json.JSONObject

class DatabaseHelper (context: Context) : SQLiteOpenHelper(context, DB_NAME, null, DB_VERSION) {
    companion object{
        private val DB_NAME = "toko"
        private val DB_VERSION = 1
        private val TABLE_BARANG = "barang"
    }

    override fun onCreate(db: SQLiteDatabase?) {
        val query = "CREATE TABLE $TABLE_BARANG (kdbrg TEXT PRIMARY KEY, nama TEXT, harga NUMERIC, expired DATE)"
        db?.execSQL(query)
    }

    override fun onUpgrade(db: SQLiteDatabase?, p1: Int, p2: Int) {
        val query = "DROP TABLE IF EXISTS $TABLE_BARANG"
        db?.execSQL(query)
        onCreate(db)
    }

    fun insert(table:String, columns:String, values:String): Boolean{
        val sql = "INSERT INTO $table ($columns) VALUES ($values)"
        try {
            val db = this.writableDatabase
            db?.execSQL(sql)
        }catch (e:java.lang.Exception){
            return false
        }
        return true
    }

    fun update(table:String, set:String, where:String): Boolean{
        val sql = "UPDATE $table SET $set WHERE $where"
        try {
            val db = this.writableDatabase
            db?.execSQL(sql)
        }catch (e:java.lang.Exception){
            return false
        }
        return true
    }

    fun delete(table:String, where:String): Boolean{
        val sql = "DELETE FROM $table WHERE $where"
        try {
            val db = this.writableDatabase
            db?.execSQL(sql)
        }catch (e:java.lang.Exception){
            return false
        }
        return true
    }

    fun select(sql: String, num_rows:Boolean=false): String {
        val db = this.readableDatabase
        val res = db.rawQuery("SELECT $sql", null)

        if(!num_rows) {
            val json = JSONArray()
            val num_cols: Int = res.getColumnCount()
            res?.moveToFirst()
            while (!res.isAfterLast()) {
                val row = JSONObject()

                for (i in 0..(num_cols - 1)) {
                    if (res.getColumnName(i) != null) {
                        try {
                            if (res.getString(i) != null) {
                                row.put(res.getColumnName(i), res.getString(i))
                            } else {
                                row.put(res.getColumnName(i), "")
                            }

                        } catch (e: Exception) {
                            e.printStackTrace();
                        }
                    }// end if
                }// end for
                json.put(row)
                res.moveToNext()
            }
            res.close()
            return json.toString()
        }else{
            // println("Num Rows: "+res.count)
            // res.count
            return res.count.toString()
        }

    }

}
